src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcTaskHost.cs

```diff
@@ -386,25 +382,19 @@ internal static string GetTaskHostNameFromHostContext(TaskHostContext hostContex
             {
                 if (s_msbuildName == null)
                 {
-                    s_msbuildName = Environment.GetEnvironmentVariable("MSBUILD_EXE_NAME");
-
-                    if (s_msbuildName == null)
-                    {
-                        s_msbuildName = "MSBuild.exe";
-                    }
+                    s_msbuildName = Environment.GetEnvironmentVariable("MSBUILD_EXE_NAME") ?? "MSBuild.exe";
```

Comment:
I'm starting to see the need for a common method that gets this information, if we don't already have one...